src/node/miner.cpp

```diff
@@ -138,9 +137,9 @@ std::unique_ptr<CBlockTemplate> BlockAssembler::CreateNewBlock(const CScript& sc
 
     int nPackagesSelected = 0;
     int nDescendantsUpdated = 0;
-    if (m_mempool) {
-        LOCK(m_mempool->cs);
-        addPackageTxs(*m_mempool, nPackagesSelected, nDescendantsUpdated);
```

Comment:
Why aren't you just changing `addPackageTxs` to use `m_mempool` directly? Seems simpler, and doesn't change the public api. Something like https://github.com/ajtowns/bitcoin/commit/944d8bc87690389de9bd51a6ea8a3223d10e4002#diff-352b412a72c7d069b77b786164c6681ee324ff0d8e61f01d48057e297db991fa

Reply:
Looking at the review discussion in #25223 introducing a check in the function and using the arrow operator feels like a step back in code quality. I also feel like having raw pointers as class members is something we should try to avoid, since its ownership semantics are ambiguous. Relying on the member variable precludes this later [comment](https://github.com/bitcoin/bitcoin/pull/25223#discussion_r898233059) on the one I linked in the PR description by making a refactor into fully static functions harder.

Reply:
I could see that being an argument for doing away with `BlockAssembler` entirely (at least as a public interface), and replacing `BlockAssembler(chainstate, mempool, options).CreateNewBlock(scriptPubKeyIn)` it with a function: `CreateNewBlock(chainstate, mempool, scriptPubKeyIn, options)` (effectively reverting #7598).

But if you're keeping `BlockAssembler` around, the entire point of that class is to assemble a block with txs from the mempool -- that's why most of its private functions require `CTxMemPool::setEntries` references, eg -- so it seems pretty strange to make it clumsier at doing that by pulling data it needs to do its job out of the class itself.

(Unrelated, but also seems a bit bogus to call the `m_last_block_*` things "members" when they're really globals...)